docs/core/testing/mstest-analyzers/mstest0038.md

```diff
@@ -0,0 +1,46 @@
+---
+title: "Don't use 'Assert.AreSame' or 'Assert.AreNotSame' with value types"
+description: "Learn about code analysis rule MSTEST0038: Don't use 'Assert.AreSame' or 'Assert.AreNotSame' with value types"
+ms.date: 01/06/2025
+f1_keywords:
+- MSTEST0038
+- AvoidAssertAreSameWithValueTypesAnalyzer
+helpviewer_keywords:
+- AvoidAssertAreSameWithValueTypesAnalyzer
+- MSTEST0038
+author: Youssef1313
+ms.author: ygerges
+---
+# MSTEST0038: Don't use 'Assert.AreSame' or 'Assert.AreNotSame' with value types
+
+| Property                            | Value                                                                  |
+|-------------------------------------|------------------------------------------------------------------------|
+| **Rule ID**                         | MSTEST0038                                                             |
+| **Title**                           | Don't use 'Assert.AreSame' or 'Assert.AreNotSame' with value types     |
+| **Category**                        | Usage                                                                  |
+| **Fix is breaking or non-breaking** | Non-breaking                                                           |
+| **Enabled by default**              | Yes                                                                    |
+| **Default severity**                | Warning                                                                |
+| **Introduced in version**           | 3.8.0                                                                  |
+| **Is there a code fix**             | No                                                                     |
+
+## Cause
+
+The use of <xref:Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreSame%2A?displayProperty=nameWithType> or <xref:Microsoft.VisualStudio.TestTools.UnitTesting.Assert.AreNotSame%2A?displayProperty=nameWithType> with one or both arguments being a value type.
```

Comment:
We don't need to escape `*` to `%2A` anymore, BTW. The only xref character to escape is `#` in the constructor APIs.